pkg/cloudprovider/aws/apis/v1alpha1/provider.go

```diff
@@ -51,6 +52,11 @@ type AWS struct {
 }
 
 func Deserialize(constraints *v1alpha5.Constraints) (*Constraints, error) {
+	if constraints.Provider == nil {
+		if constraints.Provider == nil {
```

Comment:
I'm confused - why the repeat here?

Reply:
oof my eyes must have merged those, didn't even see that... 😳